rcljava/src/main/java/org/ros2/rcljava/events/EventImpl.java

```diff
@@ -0,0 +1,164 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.ros2.rcljava.events;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.InvocationTargetException;
+
+import org.ros2.rcljava.common.JNIUtils;
+import org.ros2.rcljava.consumers.Consumer;
+import org.ros2.rcljava.events.Event;
+import org.ros2.rcljava.events.EventStatus;
+import org.ros2.rcljava.interfaces.Disposable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class serves as a bridge between ROS2's rcl_event_t and RCLJava.
+ * An Event must be created via
+ * @{link Publisher#createEvent(Class&lt;T&gt;, Consumer&lt;T&gt;)}
+ * @{link Subscription#createEvent(Class&lt;T&gt;, Consumer&lt;T&gt;)}
+ *
+ * @param <T> The status event type.
+ * @param <ParentT> The parent class type.
+ */
+public class EventImpl<
+    T extends EventStatus,
+    ParentT extends Disposable>
+implements Event<T, ParentT> {
+  private static final Logger logger = LoggerFactory.getLogger(EventImpl.class);
+
+  static {
+    try {
+      JNIUtils.loadImplementation(EventImpl.class);
+    } catch (UnsatisfiedLinkError ule) {
+      logger.error("Native code library failed to load.\n" + ule);
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param parentType The <code>Class</code> type of the parent.
+   *    It can be either a @{link org.ros2.rcljava.Publisher} or a
+   *    @{link org.ros2.rcljava.Subscription} class.
+   * @param parentReference A {@link java.lang.ref.WeakReference} to the
+   *     @{link org.ros2.rcljava.Publisher} or @{link org.ros2.rcljava.Subscription}
+   *     that created this event.
+   * @param handle A pointer to the underlying ROS 2 event structure, as an integer.
+   *     Must not be zero.
+   * @param messageType The <code>Class</code> of the messages that this
+   *     subscription will receive. We need this because of Java's type erasure,
+   *     which doesn't allow us to use the generic parameter of
+   *     @{link org.ros2.rcljava.Subscription} directly.
+   * @param callback The callback function that will be called when the event
+   *     is triggered.
+   */
+  public EventImpl(
+      final Class<ParentT> parentType,
+      final WeakReference<ParentT> parentReference,
+      final long handle,
+      final Class<T> eventStatusType,
+      final Consumer<T> callback) {
+    this.parentType = parentType;
+    this.parentReference = parentReference;
+    this.handle = handle;
+    this.eventStatusType = eventStatusType;
+    this.callback = callback;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public final Class<T> getStatusEventType() {
```

Comment:
nit: consider renaming to match the variable name
```suggestion
  public final Class<T> getEventStatusType() {
```